data_structures/queue/priority_queue.py

```
@@ -0,0 +1,273 @@
+                s += "Priority " + str(i) + ": "
+                s += str(i) + " "
+            s += "\n"
+        print(s)
+        return s
+
+
+class ElementPriorityQueue:
+    """
+    Element Priority Queue is the same as Fixed Priority Queue
+    The only difference being the value of the element itself is the priority
+    The rule for priorities are the same
+    The lowest integer has the highest priority.
+    Example:
+    priority(0) > priority(5)
+    priority(16) > priority(32)
+    You can change the priorities as per your need
+
+    >>> EPQ = ElementPriorityQueue()
+    >>> EPQ.enqueue(10)
+    >>> EPQ.enqueue(70)
+    >>> EPQ.enqueue(100)
+    >>> EPQ.enqueue(1)
+    >>> EPQ.enqueue(5)
+    >>> EPQ.enqueue(7)
+    >>> EPQ.enqueue(4)
+    >>> EPQ.enqueue(64)
+    >>> EPQ.enqueue(128)
+    >>> EPQ.print_queue()
+    [10, 70, 100, 1, 5, 7, 4, 64, 128]
+    >>> EPQ.dequeue()
+    1
+    >>> EPQ.dequeue()
+    4
+    >>> EPQ.dequeue()
+    5
+    >>> EPQ.dequeue()
+    7
+    >>> EPQ.dequeue()
+    10
+    >>> EPQ.print_queue()
+    [70, 100, 64, 128]
+    >>> EPQ.dequeue()
+    64
+    >>> EPQ.dequeue()
+    70
+    >>> EPQ.dequeue()
+    100
+    >>> EPQ.dequeue()
+    128
+    >>> EPQ.dequeue()
+    Traceback (most recent call last):
+                    ...
+    priority_queue.UnderFlow: Under Flow!
+    """
+
+    def __init__(self):
+        self.queue = []
+
+    def enqueue(self, data):
+        """
+        This function enters the element into the queue
+        If the queue is full an Exception is raised saying Over Flow!
+        """
+        if len(self.queue) == 100:
+            raise OverFlow("Over Flow!")
+        else:
+            self.queue.append(data)
+
+    def dequeue(self):
+        """
+        Return the highest priority element in FIFO order.
+        If the queue is empty then an under flow exception is raised.
+        """
+        if len(self.queue) == 0:
+            raise UnderFlow("Under Flow!")
```

Why error on trying to dequeue from an empty list? Could just return None.

---

Taking out something from an empty queue is not a permissible operation so instead of returning None I raised an exception

---

We should raise errors as @Ashley-J-George has done.  Returning None forces the caller to write code to check for errors.  Principle oF least surprise.